src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterUtilities.cs

```diff
@@ -100,5 +100,43 @@ public async Task DeleteSearchParameterAsync(RawResource searchParamResource)
                 throw customSearchException;
             }
         }
+
+        public async Task UpdateSearchParameterAsync(ITypedElement searchParam, RawResource prevSearchParamRaw)
+        {
+            try
+            {
+                var searchParameterWrapper = new SearchParameterWrapper(searchParam);
+
+                var prevSearchParam = _modelInfoProvider.ToTypedElement(prevSearchParamRaw);
+                var prevSearchParamWrapper = new SearchParameterWrapper(prevSearchParam);
```

Comment:
nit: I find it's strange to take `ITypedElement` and wrap it just for the `Url` parameter. 
`SearchParameterWrapper` does more than that, and I'm not sure it's all necessary for the purpose of getting url back.

Maybe we should have extension method for `ITypedElement` to get back url?

Reply:
I think you can also do `var searchParamUrl =  searchParam.GetStringScalar("url")` instead of creating  `searchParameterWrapper` object.

---
In reply to: [566325264](https://github.com/microsoft/fhir-server/pull/1615#discussion_r566325264) [](ancestors = 566325264)